### PR TITLE
Add story and doc folders and support

### PR DESCRIPTION
**Pull request by ingydotnet**
Stories are short essays explaining more involved YAML usage and ideas.

Docs are developer documentation.

Added build system support for building them into the site.

Also added `make stage` which publishes to:
    
`http://<gihub-user-id>-spec.yaml.io` for showing live content for pull requests.

Writing out our intent with examples will help us understand each other as we
create RFCs.

### CODE REVIEW

File: doc/glossary.md [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+* Double quoted scalar
+
+  Scalar values written in the double quote style are capable of expressing any
+  possible string value.
+  The double quoted style is the only scalar style capable of that.
+  They use a number of escape sequences to represent non-printable characters.
+
+* DOM
+
+  A DOM is an information state that is a tree of created by a composer or a
```
Comment (perlpunk) [inline doc/glossary.md:97]: "tree of ..." missing word?

Comment (ingydotnet) [reply doc/glossary.md]: fixed in 78dc8f2

**Review by perlpunk — APPROVED** [review]

File: doc/glossary.md [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+
+* Scalar
+
+  A scalar is a a leaf node that contains exactly one value.
+  Strings, numbers and booean values are examples of scalars.
+
+* Schema
+
+  Schema in YAML refers to all the external information required to process a
+  YAML information.
```
Comment (eemeli) [inline doc/glossary.md:337]:
```suggestion
  YAML document stream.
```

File: doc/glossary.md [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+
+## T
+
+* Tag
+
+  A tag is an annotation on a node.
+  Tags are identifers preceded by a `!`, like `!foo` or `!!str`.
+  A tag identifier is used to identify a function that will be applied to a
+  node when it is retrieved from the DOM.
+  The function's return type is can be considered the "type" of the node.
```
Comment (eemeli) [inline doc/glossary.md:416]:
```suggestion
  A schema may define a set of regular expression rules that allow for implicit tags to be assigned to plain scalar values.
  A tag identifier is used to identify a function that will be applied to a
  node when it is retrieved from the DOM.
  The function's return type may be considered the "type" of the node.
```

**Review by eemeli — APPROVED** [review]

File: doc/glossary.md [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+  support this.
+  Equivalent keys (if they can be detected) are not supported.
+
+* Model / Data Model
+
+  This word describes elementary kinds of data that YAML represents.
+  A YAML file is a stream of YAML documents each of which have one root node.
+  Nodes may be mappings, sequences or scalars.
+  Nodes may be annotated with anchors and/or tags.
+  An alias may be used for any node.
```
[the remark this answers is not in the document]
Comment (ingydotnet) [reply doc/glossary.md:254]: @eemeli An alias is not a node. 
A node is an actual mapping sequence or scalar.
An alias is a reference to a node but not a node itself.
By extension, you cannot put a tag or anchor on a alias.

File: doc/glossary.md (hunk 3 above)
Comment (ingydotnet) [reply doc/glossary.md:416]: @eemeli 

I added instead **While tags may be written explicitly into a YAML file using the `!abc` syntax, it is quite common for tags to be added to nodes implicitly according to the rules of a schema.**

There are many ways that a schema can resolve tag based on the values and structure of the data. Mentioning regexes here is too much detail for a glossary. We'll document that elsewhere.

File: doc/glossary.md [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+  The plain value `null` as well as the plain empty value is most often used to
+  represent it.
+
+## P
+
+* Parser / Parse
+
+  A parser is the processor in the load stack that reads tokens, matches them
+  against a grammar and write events.
+  It make also throw an error if the tokens don't match the grammar.
```
[the remark this answers is not in the document]
Comment (ingydotnet) [reply doc/glossary.md:289]: applied

File: doc/glossary.md [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+  A kind refers to the raw structure.
+  A kind should not be confused with a type.
+
+## L
+
+* Library
+
+  The DOM resolves tags to functions.
+  These functions come from the library that is registered to the DOM; often
+  the yaml standard library.
```
[the remark this answers is not in the document]
Comment (ingydotnet) [reply doc/glossary.md:212]: fixed

File: doc/glossary.md [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+
+* Information
+
+  Any of the various data bits flowing through a YAML stack.
+
+## J
+
+* JSON
+
+  When used in with a the JSON Schema or a derivative of that schema, YAML is a
```
[the remark this answers is not in the document]
Comment (ingydotnet) [reply doc/glossary.md:192]: fixed

**Pull request merged**